### PR TITLE
Making possible to edit content within the Edit Page, regardless of the portlets you have assigned.

### LINKED ISSUE #22698: Limited user without Search Portlet can't Edit Content From Edit Page Portlet
**Describe the bug**
If a limited user has permissions to edit a page and content but doesn't have in their portlets the Search Portlet, when trying to edit anything from the Edit Page Portlet an error is thrown. 

Content is able to be edited from the site browser, right clicking edit.

**To Reproduce**
1. Create a limited user, with BE permissions, edit permissions over pages and content.
2. Add only the Site Content Tool.
3. log in as the limited user.
4. Open a page
5. Click over the properties tab that shows on the right on the screen.
6. Get an error.
7. Go back to the Site Portlet
8. Right click over the page
9. Edit Properties
10. You can edit them successfully.

**Expected behavior**
You should be able to edit content within the Edit Page, regardless of the portlets you have assigned.

**Screenshots**
https://share.vidyard.com/watch/hhd62XKZsN6Z6DKNMgEmdN

Reproducible in demo (22.06) and 22.03.2 LTS.
Customer ticket https://dotcms.zendesk.com/agent/tickets/107890 and https://dotcms.zendesk.com/agent/tickets/110050

### PR DESCRIPTION
**Pull request by AndreyDotcms**
### Summary
🐛🧪🔒

Fixed a bug that allowed users to access the `edit-page` portlet without proper permissions. Added a new integration test to verify the fix. Updated the `LayoutAPIImpl` class to implement the correct permission check logic.

### Walkthrough
*  Added logic to check edit permissions for edit-page portlet ([link](https://github.com/dotCMS/core/pull/26447/files?diff=unified&w=0#diff-681b932576063cfc62ab12ef8fbeec527406db4dfefb87600e877c2f25eed671R180-R194))
*  Updated imports in `LayoutAPIImpl` class to support new logic ([link](https://github.com/dotCMS/core/pull/26447/files?diff=unified&w=0#diff-681b932576063cfc62ab12ef8fbeec527406db4dfefb87600e877c2f25eed671L7-R13))
*  Added test method to verify edit-page portlet access with valid permissions ([link](https://github.com/dotCMS/core/pull/26447/files?diff=unified&w=0#diff-da310ce30d61db7a560ee8dcf2bb9b47f0f8eb2c6f5f7ce3662d158cdd094e8bR154-R199))
*  Updated imports in `LayoutAPITest` class to support new test method ([link](https://github.com/dotCMS/core/pull/26447/files?diff=unified&w=0#diff-da310ce30d61db7a560ee8dcf2bb9b47f0f8eb2c6f5f7ce3662d158cdd094e8bL10-R18), [link](https://github.com/dotCMS/core/pull/26447/files?diff=unified&w=0#diff-da310ce30d61db7a560ee8dcf2bb9b47f0f8eb2c6f5f7ce3662d158cdd094e8bL21-R28))

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java [hunk 1]
```diff
@@ -179,6 +177,21 @@ public boolean doesUserHaveAccessToPortlet(final String portletId, final User us
 		if(loadLayoutsForUser(user).stream(). anyMatch(layout -> layout.getPortletIds().contains(portletId))){
 			return true;
 		}
+		if(portletId.equals("edit-page") ){
```
Comment (fmontes) [inline dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java:180]: Hello World

File: dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java [hunk 2]
```diff
@@ -179,6 +177,21 @@ public boolean doesUserHaveAccessToPortlet(final String portletId, final User us
...
+
+			for (final Role role: foundRoles){
+				final List<Permission> perms = permAPI.getPermissionsByRole(role, true, true);
+				for(final Permission perm : perms){
+					if((perm.getType().equals(IHTMLPage.class.getCanonicalName()) || perm.getType().equals(Contentlet.class.getCanonicalName()))
+							&& (perm.getPermission() == PermissionAPI.PERMISSION_EDIT || perm.getPermission() == 3)){
+						return true;
+					}
+				}
+			}
```
Comment (freddyDOTCMS) [inline dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java:193]: maybe is better move this code into a local method, something like

```suggestion
if(portletId.equals("edit-page") ){
    localMethod(...);
}
```

I think is a readable code

**Review by freddyDOTCMS — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java (hunk 2 above)
Comment (AndreyDotcms) [reply dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java]: Thanks for the feedback, change included https://github.com/dotCMS/core/pull/26447/commits/a389d706ea855f1c12c2c8181429517022663a4b

File: dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java [hunk 3]
```diff
@@ -171,14 +169,40 @@ public int compare(Layout l1, Layout l2) {
...
-          if(portletId==null || user==null || !user.isBackendUser()) {
-              return false;
-          }
+		if(portletId==null || user==null || !user.isBackendUser()) {
+		  return false;
+		}
 		if(loadLayoutsForUser(user).stream(). anyMatch(layout -> layout.getPortletIds().contains(portletId))){
 			return true;
 		}
+		if(portletId.equals("edit-page") ){
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java:203]: the comstant should be in the left side

Comment (AndreyDotcms) [reply dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java]: Added https://github.com/dotCMS/core/pull/26447/commits/3d8b5f6120767b293088af6db373647ddaef36b3

File: dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java [hunk 4]
```diff
@@ -171,14 +169,40 @@ public int compare(Layout l1, Layout l2) {
...
+		final List<Role> foundRoles = roleAPI.loadRolesForUser( user.getUserId(), false );
+		final PermissionAPI permAPI = APILocator.getPermissionAPI();
+
+		for (final Role role: foundRoles){
+			//get the permissions for the role
+			final List<Permission> perms = permAPI.getPermissionsByRole(role, true, true);
+			//determine if the user can edit the page
+			for(final Permission perm : perms){
+				if((perm.getType().equals(IHTMLPage.class.getCanonicalName()) || perm.getType().equals(Contentlet.class.getCanonicalName()))
+						&& (perm.getPermission() == PermissionAPI.PERMISSION_EDIT || perm.getPermission() == 3)){
```
Comment (fabrizzio-dotCMS) [inline dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java:187]: isn't there a constant for that number 3?

Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java:187]: can it be replaced by a more meaning ful const? Also That validation could be a separate functon

**Review by fabrizzio-dotCMS — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java (hunk 4 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java:187]: what is permissions 3 we have those ones:
```
final int PERMISSION_READ = 1;
	final int PERMISSION_USE = 1;
	final int PERMISSION_EDIT = 2;
	final int PERMISSION_WRITE = 2;
	final int PERMISSION_PUBLISH = 4;
	final int PERMISSION_EDIT_PERMISSIONS = 8;
	final int PERMISSION_CAN_ADD_CHILDREN = 16;
	```

Comment (AndreyDotcms) [reply dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java:187]: > what is permissions 3 we have those ones:
> 
> ```
> final int PERMISSION_READ = 1;
> 	final int PERMISSION_USE = 1;
> 	final int PERMISSION_EDIT = 2;
> 	final int PERMISSION_WRITE = 2;
> 	final int PERMISSION_PUBLISH = 4;
> 	final int PERMISSION_EDIT_PERMISSIONS = 8;
> 	final int PERMISSION_CAN_ADD_CHILDREN = 16;
> 	```
> ```

Permission 3 is when the user is capable of edit and view the content

Comment (AndreyDotcms) [reply dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java]: I made a refactor for the validations following your suggestions

**Review by fabrizzio-dotCMS — APPROVED** [review]

**Pull request merged**